homeassistant/components/dlna_dmr/media_player.py

```diff
@@ -24,209 +27,346 @@
     SUPPORT_VOLUME_MUTE,
     SUPPORT_VOLUME_SET,
 )
-from homeassistant.components.network import async_get_source_ip
-from homeassistant.components.network.const import PUBLIC_TARGET_IP
 from homeassistant.const import (
+    CONF_DEVICE_ID,
     CONF_NAME,
+    CONF_TYPE,
     CONF_URL,
-    EVENT_HOMEASSISTANT_STOP,
     STATE_IDLE,
     STATE_OFF,
     STATE_ON,
     STATE_PAUSED,
     STATE_PLAYING,
 )
 from homeassistant.core import HomeAssistant
-from homeassistant.exceptions import PlatformNotReady
-from homeassistant.helpers.aiohttp_client import async_get_clientsession
 import homeassistant.helpers.config_validation as cv
-import homeassistant.util.dt as dt_util
-
-_LOGGER = logging.getLogger(__name__)
-
-DLNA_DMR_DATA = "dlna_dmr"
-
-DEFAULT_NAME = "DLNA Digital Media Renderer"
-DEFAULT_LISTEN_PORT = 8301
+from homeassistant.helpers.device_registry import CONNECTION_UPNP
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    CONF_CALLBACK_URL_OVERRIDE,
+    CONF_LISTEN_PORT,
+    CONF_POLL_AVAILABILITY,
+    DEFAULT_NAME,
+    LOGGER as _LOGGER,
+)
+from .data import EventListenAddr, get_domain_data
 
+# Configuration via YAML is deprecated in favour of config flow
 CONF_LISTEN_IP = "listen_ip"
-CONF_LISTEN_PORT = "listen_port"
-CONF_CALLBACK_URL_OVERRIDE = "callback_url_override"
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_URL): cv.string,
-        vol.Optional(CONF_LISTEN_IP): cv.string,
-        vol.Optional(CONF_LISTEN_PORT, default=DEFAULT_LISTEN_PORT): cv.port,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_CALLBACK_URL_OVERRIDE): cv.url,
-    }
+PLATFORM_SCHEMA = vol.All(
+    cv.deprecated(CONF_URL),
```

Comment:
Should the whole domain be marked as deprecated?

Reply:
Has my preference. By the looks of it all options can be set via the OptionsFlow.

Reply:
The `dlna_dmr` yaml config is done as a platform within the `media_player` domain, and gets used like so:
```
media_player:
  - platform: dlna_dmr
    url: http://fake/fake.xml
```

I'm not sure how to mark this as deprecated, since `cv.deprecated` only applies to keys, not key/value combos.

Reply:
I think we have done a logger message with level of warning that it's deprecated advising removal of the yaml config in the past for this case.

Reply:
Example #50312

Reply:
Thanks for the example, and done.